lib/ansible/module_utils/redfish_utils.py

```diff
@@ -2263,3 +2263,91 @@ def get_manager_health_report(self, manager_uri):
 
     def get_multi_manager_health_report(self):
         return self.aggregate_managers(self.get_manager_health_report)
+
+    def set_manager_nic(self, nic_addr, nic_config):
+        # Get EthernetInterface collection
+        response = self.get_request(self.root_uri + self.manager_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+        if 'EthernetInterfaces' not in data:
+            return {'ret': False, 'msg': "EthernetInterfaces resource not found"}
+        ethernetinterfaces_uri = data["EthernetInterfaces"]["@odata.id"]
+        response = self.get_request(self.root_uri + ethernetinterfaces_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+        uris = [a.get('@odata.id') for a in data.get('Members', []) if
+                a.get('@odata.id')]
+
+        # Find target EthernetInterface
+        target_ethernet_uri = None
+        target_ethernet_current_setting = None
+        if nic_addr == 'null':
+            # Find root_uri matched EthernetInterface when nic_addr is not specified
+            nic_addr = (self.root_uri).split('/')[-1]
```

Comment:
The `root_uri` may include a port number (e.g. 192.168.1:8000). Should probably split that off also.

Reply:
Thanks for reminder, code is updated.